robusta_krr/core/integrations/kubernetes.py

```diff
@@ -109,28 +115,47 @@ async def __build_obj(
             pods=await self.__list_pods(item),
         )
 
+    async def _list_resources(self, all_namespaces_fn, namespaced_fn):
+        resources = []
+        if self.config.namespaces == "*":
+            ret: V1DeploymentList = await asyncio.to_thread(all_namespaces_fn, watch=False)
+            for item in ret.items:
+                resources.append(item)
+        else:
+            for namespace in self.config.namespaces:
```

Comment:
Here we are doing a for loop in a blocking manner. Using `asyncio.gather` should give us a performance boost